CMakeLists.txt

```diff
@@ -0,0 +1,377 @@
+    lib/src/ast_filter.c
+    lib/src/ast_float.c
+    lib/src/ast_foreach.c
+    lib/src/ast_function_name.c
+    lib/src/ast_identifier.c
+    lib/src/ast_index_name.c
+    lib/src/ast_integer.c
+    lib/src/ast_label.c
+    lib/src/ast_labels_operator.c
+    lib/src/ast_line_comment.c
+    lib/src/ast_list_comprehension.c
+    lib/src/ast_load_csv.c
+    lib/src/ast_map.c
+    lib/src/ast_map_projection.c
+    lib/src/ast_map_projection_all_properties.c
+    lib/src/ast_map_projection_identifier.c
+    lib/src/ast_map_projection_literal.c
+    lib/src/ast_map_projection_property.c
+    lib/src/ast_map_projection_selector.c
+    lib/src/ast_match.c
+    lib/src/ast_match_hint.c
+    lib/src/ast_merge.c
+    lib/src/ast_merge_action.c
+    lib/src/ast_merge_properties.c
+    lib/src/ast_named_path.c
+    lib/src/ast_node_id_lookup.c
+    lib/src/ast_node_index_lookup.c
+    lib/src/ast_node_index_query.c
+    lib/src/ast_node_pattern.c
+    lib/src/ast_none.c
+    lib/src/ast_null.c
+    lib/src/ast_on_create.c
+    lib/src/ast_on_match.c
+    lib/src/ast_order_by.c
+    lib/src/ast_parameter.c
+    lib/src/ast_pattern.c
+    lib/src/ast_pattern_comprehension.c
+    lib/src/ast_pattern_path.c
+    lib/src/ast_proc_name.c
+    lib/src/ast_profile_option.c
+    lib/src/ast_projection.c
+    lib/src/ast_prop_name.c
+    lib/src/ast_property_operator.c
+    lib/src/ast_query.c
+    lib/src/ast_query_clause.c
+    lib/src/ast_query_option.c
+    lib/src/ast_range.c
+    lib/src/ast_reduce.c
+    lib/src/ast_rel_id_lookup.c
+    lib/src/ast_rel_index_lookup.c
+    lib/src/ast_rel_index_query.c
+    lib/src/ast_rel_pattern.c
+    lib/src/ast_reltype.c
+    lib/src/ast_remove.c
+    lib/src/ast_remove_item.c
+    lib/src/ast_remove_labels.c
+    lib/src/ast_remove_property.c
+    lib/src/ast_return.c
+    lib/src/ast_schema_command.c
+    lib/src/ast_set.c
+    lib/src/ast_set_all_properties.c
+    lib/src/ast_set_item.c
+    lib/src/ast_set_labels.c
+    lib/src/ast_set_property.c
+    lib/src/ast_shortest_path.c
+    lib/src/ast_single.c
+    lib/src/ast_slice_operator.c
+    lib/src/ast_sort_item.c
+    lib/src/ast_start.c
+    lib/src/ast_start_point.c
+    lib/src/ast_statement.c
+    lib/src/ast_statement_option.c
+    lib/src/ast_string.c
+    lib/src/ast_subscript_operator.c
+    lib/src/ast_true.c
+    lib/src/ast_unary_operator.c
+    lib/src/ast_union.c
+    lib/src/ast_unwind.c
+    lib/src/ast_using_index.c
+    lib/src/ast_using_join.c
+    lib/src/ast_using_periodic_commit.c
+    lib/src/ast_using_scan.c
+    lib/src/ast_with.c
+    lib/src/errors.c
+    lib/src/operators.c
+    lib/src/parser.c
+    lib/src/parser_config.c
+    lib/src/result.c
+    lib/src/segment.c
+    lib/src/string_buffer.c
+    lib/src/util.c
+    lib/src/vector.c)
+
+add_library (libcypher-parser ${SOURCES})
+target_compile_options (libcypher-parser PUBLIC /wd4267 /wd4244 /wd4047)
+
+set (TESTS)
+
+list (APPEND TESTS
+    lib/test/check_call.c
```

Comment:
This list duplicates the list in lib/test/Makefile.am. Is there any way to avoid that duplication?

Reply:
Yes, I will add a macro in `CMakeLists.txt` to parse the source files list out of `lib/test/Makefile.am`.